internal/mesh/internal/controllers/sidecarproxy/fetcher/data_fetcher.go

```diff
@@ -12,13 +16,23 @@ import (
 	pbcatalog "github.com/hashicorp/consul/proto-public/pbcatalog/v1alpha1"
 	pbmesh "github.com/hashicorp/consul/proto-public/pbmesh/v1alpha1"
 	"github.com/hashicorp/consul/proto-public/pbresource"
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
 )
 
 type Fetcher struct {
-	Client pbresource.ResourceServiceClient
-	Cache  *sidecarproxycache.Cache
+	Client            pbresource.ResourceServiceClient
+	DestinationsCache *sidecarproxycache.DestinationsCache
+	ProxyCfgCache     *sidecarproxycache.ProxyConfigurationCache
+}
+
+func New(client pbresource.ResourceServiceClient,
```

Comment:
```suggestion
func NewFetcher(client pbresource.ResourceServiceClient,
```

Reply:
given that it's in its own package, I think it's nice to be able to say `fetcher.New` instead of `fetcher.NewFetcher`